presto-orc/src/main/java/com/facebook/presto/orc/reader/ByteStreamReader.java

```diff
@@ -173,12 +174,180 @@ public String toString()
     public void close()
     {
         systemMemoryContext.close();
-        nullVector = null;
     }
 
     @Override
     public long getRetainedSizeInBytes()
     {
-        return INSTANCE_SIZE + sizeOf(nullVector);
+        return INSTANCE_SIZE;
+    }
+
+    @Override
+    public void setFilterAndChannel(Filter filter, int channel, int columnIndex, Type type)
+    {
+        checkArgument(type == TINYINT, "Unsupported type: " + type.getDisplayName());
+        super.setFilterAndChannel(filter, channel, columnIndex, type);
+    }
+
+    @Override
+    public void scan()
+            throws IOException
+    {
+        if (!rowGroupOpen) {
+            openRowGroup();
+        }
+        beginScan(presentStream, null);
+        QualifyingSet input = inputQualifyingSet;
+        QualifyingSet output = outputQualifyingSet;
+        int end = input.getEnd();
+        int rowsInRange = end - posInRowGroup;
+        int valuesSize = end;
+        int[] inputPositions = input.getPositions();
+        if (filter != null) {
+            output.reset(rowsInRange);
+        }
+        if (values == null || values.length < valuesSize) {
+            values = new byte[valuesSize];
+        }
+
+        if (dataStream == null) {
+            processAllNulls();
+            endScan(presentStream);
+            return;
+        }
+
+        int nextActive = inputPositions[0];
+        int activeIdx = 0;
+        int toSkip = 0;
+        for (int i = 0; i < rowsInRange; i++) {
+            if (i + posInRowGroup == nextActive) {
+                if (nextActive == truncationRow) {
+                    break;
+                }
+                if (presentStream != null && !present[i]) {
```

Comment:
> It's not clear to me when presentStream is null or not

`presentStream == null` means that all values are non-null. This code is pre-Aria. We don't want to use Optional here because of extra cost of `Optional.get()` for each value.